mlabs/src/Mlabs/EfficientNFT/Types.hs

```diff
@@ -81,31 +86,57 @@ data ChangeOwnerParams = ChangeOwnerParams
 type GenericContract a = forall w s. Contract w s Text a
 type UserContract a = forall s. Contract (Last NftId) s Text a
 
-data OwnerData = OwnerData
-  { odOwnerPkh :: !PaymentPubKeyHash
-  , odPrice :: !Natural
-  }
-  deriving stock (Hask.Show)
-
-PlutusTx.makeLift ''OwnerData
-PlutusTx.unstableMakeIsData ''OwnerData
-
-data PlatformConfig = PlatformConfig
-  { pcMarketplacePkh :: !PaymentPubKeyHash
-  , -- | % share of the marketplace multiplied by 100
-    pcMarketplaceShare :: !Natural
-  }
-  deriving stock (Hask.Show)
-
-PlutusTx.makeLift ''PlatformConfig
-PlutusTx.unstableMakeIsData ''PlatformConfig
-
 data MintAct
-  = MintToken OwnerData
-  | ChangePrice OwnerData Natural
-  | ChangeOwner OwnerData PaymentPubKeyHash
+  = MintToken NftId
+  | ChangePrice NftId Natural
+  | ChangeOwner NftId PaymentPubKeyHash
+  | BurnToken NftId
   deriving stock (Hask.Show)
 
 PlutusTx.unstableMakeIsData ''MintAct
 
 type ContentHash = BuiltinByteString
+
+class Hashable a where
```

Comment:
lawless typeclass but I suppose this is a good one to have to ensure everything is hashed the same way in all places